pkg/kubelet/cm/dra/manager.go

```diff
@@ -108,30 +109,21 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 			continue
 		}
 
-		claimInfo := m.cache.get(*claimName, pod.Namespace)
-		if claimInfo == nil {
-			// claim does not exist in cache, create new claimInfo object
-			// to be processed later.
-			claimInfo = newClaimInfoFromResourceClaim(resourceClaim)
-		}
-
-		// We delay checkpointing of this change until this call
-		// returns successfully. It is OK to do this because we
-		// will only return successfully from this call if the
-		// checkpoint has succeeded. That means if the kubelet is
-		// ever restarted before this checkpoint succeeds, the pod
-		// whose resources are being prepared would never have
-		// started, so it's OK (actually correct) to not include it
-		// in the cache.
-		claimInfo.addPodReference(pod.UID)
-
-		if claimInfo.prepared {
-			// Already prepared this claim, no need to prepare it again
-			continue
+		claimInfo, ok := m.cache.get(*claimName, pod.Namespace)
+		if ok {
+			// Claim exists in cache. Reference the pod.
+			m.cache.addPodReference(*claimName, pod.Namespace, pod.UID)
+			if claimInfo.prepared {
+				// Already prepared this claim, no need to prepare it again
+				continue
+			}
 		}
+		// This saved claim will be used to update ClaimInfo cache
+		// after NodePrepareResources GRPC succeeds
+		resourceClaims[resourceClaim.UID] = resourceClaim
```

Comment:
If we do what I suggest above about removing `get()` then this turns into:

```suggestion
// Save a reference to the claim so we can add it to the ClaimInfo
// cache after the NodePrepareResources gRPC call succeeds
resourceClaims[resourceClaim.UID] = resourceClaim

// If the cache already contains info about this claim, then add a pod reference to it
if m.cache.Contains(*claimName, pod.Namespace) {
    m.cache.addPodReference(*claimName, pod.Namespace, pod.UID)
}

// If a claim is already prepared, there is no need to prepare it again
if m.cache.IsPrepared(*claimName, pod.Namespace) {
	continue
}
```

Reply:
It's the same as a bit changed current code:
```
claimInfo, ok := m.cache.get(*claimName, pod.Namespace)
if ok {
        m.cache.addPodReference(*claimName, pod.Namespace, pod.UID)
}

if claimInfo.prepared {
    continue
}

// I intentionally do it here, after the above check
resourceClaims[resourceClaim.UID] = resourceClaim 
```

However, this code is a bit less clear than the current code as it checks ```prepared``` field for no reason if the claimInfo is not in the cache.

BTW, I agree that getting rid of ```cache.get``` makes sense, but for another reason.